pennylane_qiskit/converter.py

```diff
@@ -35,17 +35,80 @@
 dagger_map = {"SdgGate": qml.S, "TdgGate": qml.T, "SXdgGate": qml.SX}
 
 
-def _check_parameter_bound(param: Parameter, var_ref_map: Dict[Parameter, Any]):
+def _check_parameter_bound(param: Parameter, trainable_params: Dict[Parameter, Any]):
     """Utility function determining if a certain parameter in a QuantumCircuit has
     been bound.
 
     Args:
         param (qiskit.circuit.Parameter): the parameter to be checked
-        var_ref_map (dict[qiskit.circuit.Parameter, Any]):
+        trainable_params (dict[qiskit.circuit.Parameter, Any]):
             a dictionary mapping qiskit parameters to trainable parameter values
     """
-    if isinstance(param, Parameter) and param not in var_ref_map:
-        raise ValueError("The parameter {} was not bound correctly.".format(param))
+    if isinstance(param, Parameter) and param not in trainable_params:
+        raise ValueError(f"The parameter {param} was not bound correctly.".format(param))
+
+
+def _format_params_dict(quantum_circuit, params, *args, **kwargs):
+    """Processes the inputs for calling the quantum function and returns
+    a dictionary of the format {Parameter("name"): value} for all the parameters.
+
+    For a ``quantum_circuit`` with parameters ``[Parameter("phi"), Parameter("psi"), Parameter("theta")]``,
+    inputs can be one of the following:
+        1. the kwargs passed to when calling the qfunc, other than ``params`` and ``wires``.
+            The keys in kwargs are expected to correspond to the names of the Parameters on
+            the QuantumCircuit, i.e. (qc, None, phi=0.35, theta=0.2, psi=1.7)
+        2. the args passed when calling the qfunc, assigned to the Parameters in alphabetical
+            order, i.e. (qc, None, 0.35, 1.7, 0.2)
+        3. Some combination of args and kwargs, i.e. (qc, None, 0.35, 0.2, psi=1.7)
+        4. (legacy) ``params`` from the kwarg ``params`` of the qfunc call, which is expected
+            to already be a dictionary of the format {Parameter("name"): value}, i.e.
+            (qc, {Parameter("phi"): 0.35, Parameter("psi"): 1.7, Parameter("theta"): 0.2})
+        5. (legacy) ``params`` passed as a single arg, which is expected
+            to already be a dictionary of the format {Parameter("name"): value}, i.e.
+            (qc, None, {Parameter("phi"): 0.35, Parameter("psi"): 1.7, Parameter("theta"): 0.2})
+
+    Returns:
+        params (dict): A dictionary mapping ``quantum_circuit.parameters`` to values
+    """
+
+    # if nothing passed to params, and a dictionary has been passed as a single argument, then assume it is params
+    if params is None and (len(args) == 1 and isinstance(args[0], dict)):
+        params = args[0]
```

Comment:
Yes, we absolutely can - in theory the next condition should never be fulfilled if this condition is fulfilled, and it will just go to the return at the end of the function. Just adding the return here seems safer and easier to read.